Add rubocop group to Gemfile for use in CI

eugeneius opened the pull request:
Since https://github.com/rails/rails/pull/38834 we install the latest version of RuboCop in our GitHub Actions workflow for speed, but this sacrifices reproducibility; the results will change whenever RuboCop publishes a new version. Instead we can add a new group to our Gemfile that just contains the dependencies necessary to run RuboCop, and skip installing everything else in CI.

Unfortunately it's not possible to tell Bundler to only install gems from a single group, so we have to tell it not to install every other group instead.

kaspth commented:
Hm, it says our cache size is 500MB:
https://github.com/rails/rails/pull/38836/checks?check_run_id=540858669#step:4:9

eugeneius committed:
Add rubocop group to Gemfile for use in CI

Since c1e7268c83336777655b20f9e23892d8143c0243 we install the latest
version of RuboCop in our GitHub Actions workflow for speed, but this
sacrifices reproducibility; the results will change whenever RuboCop
publishes a new version. Instead we can add a new group to our Gemfile
that just contains the dependencies necessary to run RuboCop, and skip
installing everything else in CI.

Unfortunately it's not possible to tell Bundler to only install gems
from a single group, so we have to tell it not to install every other
group instead.

kaspth commented:
Cool, looks like switching the cache key to `-rubocop-` fixed that.

eugeneius commented:
That last build ran without a cache, I triggered one more build to see how it works when one exists. Maybe we don't even need a cache!

eugeneius commented:
<img width="685" alt="Screenshot 2020-03-27 at 23 57 43" src="https://user-images.githubusercontent.com/432189/77809239-c7e7b080-7086-11ea-83ad-bc7bdfd0212b.png">

🐎

kaspth commented:
lol, looks like it's fairly neck and neck. I say we keep it just in case, but your call.

eugeneius commented:
I think it's worth keeping, let's save Ruby Central some bandwidth costs 😄

simi commented on .github/workflows/rubocop.yml:
```diff
@@ -12,8 +12,17 @@ jobs:
+          ${{ runner.os }}-rubocop-
+    - name: Install gems
       run: |
-        gem install --no-document rubocop rubocop-performance rubocop-rails
```
Wouldn't be easier to specify particular versions in here?

kaspth replied:
The cache time and gem install is down to a combined 2 seconds. It’s far better that Gemfile.lock is the source of truth. Appreciate the suggestion but I think we’re settled on this 😄🙏
